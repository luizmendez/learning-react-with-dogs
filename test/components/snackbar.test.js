import React from 'react';
import Snackbar from 'components/snackbar/snackbar/';

jest.useFakeTimers();

const noop = () => {};

const props = {
    message: '',
    type: '',
    msToClose: 0,
    callback: noop,
    removeMessage: noop
};

describe('<Snackbar />', () => {
    it('renders the Snackbar component', () => {
        const wrapper = shallow(<Snackbar message="Holis World" />);
        expect(wrapper.find('.snackbar').length).toBe(1);
        expect(wrapper.text()).toMatch('Holis World');
        expect(wrapper).toMatchSnapshot();
    });

    it('should render a child component', () => {
        const wrapper = shallow(
            <Snackbar>
                <span>Holis</span>
            </Snackbar>
        );
        expect(wrapper.find('span').length).toBe(1);
        expect(wrapper.find('span').text()).toMatch('Holis');
        expect(wrapper).toMatchSnapshot();
    });

    it('applies the given type as a css class', () => {
        const wrapper = shallow(<Snackbar message="Holis World" type="warning" />);
        expect(wrapper.find('.alert-warning').length).toBe(1);
        expect(wrapper).toMatchSnapshot();
    });

    it('should not render if there is no message', () => {
        const wrapper = shallow(<Snackbar {...props} />);
        expect(wrapper.find('.snackbar').length).toBe(0);
        expect(wrapper).toMatchSnapshot();
    });

    it('should not render if close button is clicked', () => {
        const wrapper = shallow(<Snackbar message="Holis" callback={noop} />);
        const closeBtn = wrapper.find('button');
        expect(wrapper.find('.snackbar').length).toBe(1);
        closeBtn.simulate('click');
        expect(wrapper.find('.snackbar').length).toBe(0);
        expect(wrapper).toMatchSnapshot();
    });

    it('should call callback if exists after closing', () => {
        const mockCallback = jest.fn();
        const wrapper = shallow(<Snackbar message="Holis" callback={mockCallback} />);
        const closeBtn = wrapper.find('button');
        closeBtn.simulate('click');
        expect(mockCallback).toBeCalledTimes(1);
        expect(wrapper).toMatchSnapshot();
    });

    /*** Test with problem calling callback when advacing time, tested in UI and working ***/
    //it('should close snackbar after 3 seconds', () => {
    //    const wrapper = shallow(<Snackbar message="Holis" callback={noop} msToClose={3000} />);
    //    expect(wrapper.find('.snackbar').length).toBe(1);
    //    jest.advanceTimersByTime(3001);
    //    expect(wrapper.find('.snackbar').length).toBe(0);
    //});

    it('should not show close button', () => {
        const wrapper = shallow(<Snackbar message="Holis" showClose={false} />);
        expect(wrapper.find('.btn').length).toBe(0);
    });
});
