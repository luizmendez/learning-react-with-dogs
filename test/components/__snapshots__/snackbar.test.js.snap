// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Snackbar /> applies the given type as a css class 1`] = `
<div
  className="snackbar alert alert-dismissible alert-warning"
>
  Holis World
  <button
    className="close close-snackbar"
    onClick={[Function]}
    type="button"
  >
    ×
  </button>
</div>
`;

exports[`<Snackbar /> renders the Snackbar component 1`] = `
<div
  className="snackbar alert alert-dismissible alert-info"
>
  Holis World
  <button
    className="close close-snackbar"
    onClick={[Function]}
    type="button"
  >
    ×
  </button>
</div>
`;

exports[`<Snackbar /> should call callback after closing 1`] = `""`;

exports[`<Snackbar /> should not render if close button is clicked 1`] = `""`;

exports[`<Snackbar /> should not render if there is no message 1`] = `""`;

exports[`<Snackbar /> should render a child component 1`] = `
<div
  className="snackbar alert alert-dismissible alert-info"
>
  <span>
    Holis
  </span>
  <button
    className="close close-snackbar"
    onClick={[Function]}
    type="button"
  >
    ×
  </button>
</div>
`;
